File: argo/read.jl
using CSV, DataFrames

dpath = "/home/kshedden/data/Teaching/argo/julia"

lat = open(joinpath(dpath, "lat.csv.gz")) do io
    CSV.read(io, DataFrame)
end
lat = lat[:, 1]

lon = open(joinpath(dpath, "lon.csv.gz")) do io
    CSV.read(io, DataFrame)
end
lon = lon[:, 1]

date = open(joinpath(dpath, "date.csv.gz")) do io
    CSV.read(io, DataFrame)
end
date = date[:, 1]
date = [x.value for x in date - minimum(date)]

pressure = open(joinpath(dpath, "pressure.csv.gz")) do io
    CSV.read(io, DataFrame)
end
pressure = pressure[:, 1]

temp = open(joinpath(dpath, "temp.csv.gz")) do io
    CSV.read(io, DataFrame)
end
temp = Matrix(temp)

psal = open(joinpath(dpath, "psal.csv.gz")) do io
    CSV.read(io, DataFrame)
end
psal = Matrix(psal)

File: argo/read.R
library(readr)
library(dplyr)
library(lubridate)

# There is no R version of prep.py, you must run the
# Python or julia prep script before using read.R.

# This path must match the path in prep.py.
qpath = "/home/kshedden/data/Teaching/argo/python"

# The latitude where each profile was recorded
lat = read_csv(file.path(qpath, "lat.csv.gz"))
lat = as.vector(lat[,1])$Column1

# The longitude where each profile was recorded
lon = read_csv(file.path(qpath, "lon.csv.gz"))
lon = as.vector(lon[,1])$Column1

# The day of year on which a profile was recorded
date = read_csv(file.path(qpath, "date.csv.gz"))
date = as.vector(date[,1])$Column1
day = yday(date)

# The pressure (in decibars) for each point along a profile.
pressure = read_csv(file.path(qpath, "pressure.csv.gz"))
pressure = as.vector(pressure[,1])$Column1

# The temperature profiles
temp = read_csv(file.path(qpath, "temp.csv.gz"), col_names=F)
temp = as.matrix(temp) %>% t

# The salinity profiles
psal = read_csv(file.path(qpath, "psal.csv.gz"), col_names=F)
psal = as.matrix(psal) %>% t
